### ISSUE TITLE
error on accessing node.frame while applying scale anim

### ISSUE THREAD

**Issue by simoami** [facebook/AsyncDisplayKit#2233]
Hello,
I'm applying a facebook pop scale anim `POPSpringAnimation(propertyNamed: kPOPViewScaleXY)` to an ASButtonNode. The effect is applied on touch up eveny. If the effect is applied sequentially in a few seconds intervals, the effect works great each time. But if an effect applied overlaps an existing one in progress, though fast taps, I hit the following error:

``` obj-c
ASDisplayNodeAssert(
  CATransform3DIsIdentity(self.transform), 
  @"-[ASDisplayNode frame] - self.transform must be identity in order to use the frame property.  (From Apple's UIView documentation: If the transform property is not the identity transform, the value of this property is undefined and therefore should be ignored.)"
);
```

My code is as follows:

``` swift
let favoriteButton = ASButtonNode()
func setup() {
    favoriteButton.addTarget(self, action: #selector(self.favoriteButtonPressed), forControlEvents: .TouchUpInside)
}
func favoriteButtonPressed() {
    //favoriteButton.pop_removeAllAnimations()
    //favoriteButton.transform = CATransform3DIdentity
    favoriteButton.pop_addAnimation(getBounceAnim(), forKey: "bounce")
}
func getBounceAnim() -> POPPropertyAnimation {
    let spring = POPSpringAnimation(propertyNamed: kPOPViewScaleXY)
    spring.fromValue = NSValue(CGPoint: CGPointMake(1.5, 1.5))
    spring.toValue = NSValue(CGSize: CGSizeMake(1, 1))
    spring.springBounciness = 15 // a float between 0 and 20
    spring.springSpeed = 10
    spring.beginTime = CACurrentMediaTime()
    return spring
}

```

Any ideas how I can avoid the assert?

**Comment by simoami** [comment]
It seems that the issue is caused from accessing the frame property in https://github.com/facebook/AsyncDisplayKit/blob/1.9.90/AsyncDisplayKit/Details/_ASDisplayViewAccessiblity.mm#L88

```
      [accessiblityElement setAccessibilityFrame:UIAccessibilityConvertFrameToScreenCoordinates(subnode.frame, self)];
```
which then calls:

```
- (CGRect)frame
{
  _bridge_prologue_read;

  // Frame is only defined when transform is identity.
#if DEBUG
  // Checking if the transform is identity is expensive, so disable when unnecessary. We have assertions on in Release, so DEBUG is the only way I know of.
  ASDisplayNodeAssert(CATransform3DIsIdentity(self.transform), @"-[ASDisplayNode frame] - self.transform must be identity in order to use the frame property.  (From Apple's UIView documentation: If the transform property is not the identity transform, the value of this property is undefined and therefore should be ignored.)");
#endif
...
}
```

But it seems that this now has changed in Master as you're now accessing the node bounds instead of the frame: https://github.com/facebook/AsyncDisplayKit/blob/master/AsyncDisplayKit/Details/_ASDisplayViewAccessiblity.mm#L70-L71

Can you confirm this was intended to fix this issue?

**Comment by simoami** [comment]
Update: This works fine in master after switching to .bounds instead of .frame: https://github.com/facebook/AsyncDisplayKit/blob/master/AsyncDisplayKit/Details/_ASDisplayViewAccessiblity.mm#L70-L71

I can confirm the issue is fixed.

**Comment by bawn** [comment]
+1

**Comment by maicki** [comment]
@simoami Hey, is this issue fixed with 2.0? Can we close it? Thanks!

**Comment by hannahmbanana** [comment]
@simoami @bawn - thanks for confirming the fix! Let us know if you run into any other issues with this.
